editoast/src/views/v2/train_schedule.rs

```diff
@@ -682,41 +683,104 @@ pub async fn project_path(
 #[derive(Debug, Serialize, ToSchema)]
 enum SimulationSummaryResultResponse {
     // Minimal information on a simulation's result
-    #[allow(dead_code)]
     Success {
         // Length of a path
         length: u64,
         // travel time
-        time: f64,
+        time: u64,
         energy_consumption: f64,
     },
-    // Pathfinding not found for a train id
-    #[allow(dead_code)]
-    PathfindingNotFound,
-    // Running time not found or a train id
-    #[allow(dead_code)]
-    RunningTimeNotFound,
+    // Pathfinding fails for a train id
+    PathfindingFailed,
+    // Running time fails or a train id
+    RunningTimeFailed,
 }
 
 /// Retrieve simulation information for a given train list.
 /// Useful for finding out whether pathfinding/simulation was successful.
 #[utoipa::path(
     tag = "train_schedulev2",
+    params(SimulationBatchParams),
     responses(
-        (status = 200, description = "Project Path Output", body = HashMap<Identifier, SimulationSummaryResultResponse>),
+        (status = 200, description = "Project Path Output", body = HashMap<i64, SimulationSummaryResultResponse>),
     ),
 )]
 #[get("/simulation_summary")]
 pub async fn simulations_summary(
-    _db_pool: Data<DbPool>,
-    _redis_client: Data<RedisClient>,
-    _params: QsQuery<ProjectPathParams>,
-    _core_client: Data<CoreClient>,
-    _infra_id: Query<i64>,
-) -> Result<Json<HashMap<Identifier, SimulationSummaryResultResponse>>> {
-    // TO DO
-    // issue:x https://github.com/OpenRailAssociation/osrd/issues/6857
-    Ok(Json(HashMap::new()))
+    db_pool: Data<DbPool>,
+    redis_client: Data<RedisClient>,
+    params: QsQuery<SimulationBatchParams>,
+    core_client: Data<CoreClient>,
+) -> Result<Json<HashMap<i64, SimulationSummaryResultResponse>>> {
```

Comment:
Can you explain why you replace the `Identifier` with a `i64`? I don't have any context... but just looking at that, it feels like a regression in terms of typing.

Reply:
The keys in our HashMap are train ids (i64)

Reply:
Ok, so we discussed it offline and `Identifier` is also specifically for `String`, not integers. We (in `editoast` do not have (yet?) an `Identifier` type allowing to wrap integers.

So now I understand this change. I'm resolving the comment.